packages/react-theme/src/themeToTokensObject.ts

```diff
@@ -0,0 +1,10 @@
+import { Theme } from './types';
+
+export function themeToTokensObject<TTheme extends Theme>(theme: TTheme): Record<keyof TTheme, string> {
```

Comment:
I was thinking why `tokens` are hardcoded instead of being exported as `themeToTokensObject(webLightTheme)`.

That would reduce the bundle size by not having hardcoded tokens but also add a theme (`webLightTheme`) which might not be used in the application.

Another problem is that with a built-in theme, app engineer does not know whether they should import `tokens` or rather call this function. Would it make sense to rename it to something like `customThemeToTokensObject` or at least add a comment to describe when this should (not) be used?

Reply:
Compared to `tokens` approach it's not tree shakable as bundlers do not know the shape of our output.

[Rollup REPL](https://rollupjs.org/repl/?version=2.67.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHRva2Vuc0ElMjBmcm9tJTIwJy4lMkZ0b2tlbnNBJyU1Q25pbXBvcnQlMjAlN0IlMjB0b2tlbnNCJTIwJTdEJTIwZnJvbSUyMCcuJTJGdG9rZW5zQiclNUNuJTVDbmNvbnNvbGUubG9nKHRva2Vuc0EuY29sb3IpJTVDbmNvbnNvbGUubG9nKHRva2Vuc0IuY29sb3IpJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIydG9rZW5zQS5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGNvbG9yJTIwJTNEJTIwJ3ZhcigtLWNvbG9yKSclMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIydG9rZW5zQi5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJmdW5jdGlvbiUyMHRoZW1lVG9Ub2tlbnModGhlbWUpJTIwJTdCJTVDbiUyMCUyMGNvbnN0JTIwdG9rZW5zJTIwJTNEJTIwJTdCJTdEJTNCJTVDbiUyMCUyMGNvbnN0JTIwa2V5cyUyMCUzRCUyME9iamVjdC5rZXlzKHRoZW1lKSUzQiU1Q24lMjAlMjAlNUNuJTVDdGZvciUyMChjb25zdCUyMGtleSUyMG9mJTIwa2V5cyklMjAlN0IlNUNuJTIwJTIwJTIwJTIwdG9rZW5zJTVCa2V5JTVEJTIwJTNEJTIwJTYwdmFyKC0tJTI0JTdCa2V5JTdEKSU2MCUzQiU1Q24lMjAlMjAlN0QlNUNuJTVDdCU1Q24lMjAlMjByZXR1cm4lMjB0b2tlbnMlM0IlNUNuJTdEJTVDbiU1Q25leHBvcnQlMjBjb25zdCUyMHRva2Vuc0IlMjAlM0QlMjB0aGVtZVRvVG9rZW5zKCU3QiUyMGNvbG9yJTNBJTIwJ3JlZCclMjAlN0QpJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

Reply:
Added a comment to the function to describe its pros and cons